repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -2710,7 +2715,27 @@ public List<String> propagateClassification(String entityGuid, String classifica
                 throw new AtlasBaseException(String.format("propagateClassification(entityGuid=%s, classificationVertexId=%s): classification vertex not found", entityGuid, classificationVertexId));
             }
 
-            List<AtlasVertex> impactedVertices = entityRetriever.getIncludedImpactedVerticesV2(entityVertex, relationshipGuid, classificationVertexId);
+            /*
+                If propagateThroughLineage was true at past then updated to false we need to delete the propagated
+             */
+            Boolean updatedRestrictPropagationThroughLineage = entityRetriever.toAtlasClassification(classificationVertex).getRestrictPropagationThroughLineage();
```

Comment:
instead of converting to AtlasClassification, fetch just required property value from the vertex
converting to AtlasClassification will read all properties from vertex where as we just need single property value